torch/quantization/ns/graph_matcher.py

```diff
@@ -111,27 +147,52 @@ def __init__(
     def __iter__(self):
         return self
 
-    def __next__(self) -> Node:
+    def __next__(self) -> Tuple[Node, Node]:
         """
-        Returns the next matchable node.
+        Returns the next matchable subgraph, defined by (start_node, end_node)
         """
         while len(self.stack) > 0:
-            cur_node = self.stack.pop()
-            if cur_node in self.seen_nodes:
+            cur_end_node = self.stack.pop()
+            if cur_end_node in self.seen_nodes:
                 continue
-            self.seen_nodes.add(cur_node)
+
+            # for subgraphs which are single nodes, start_node == end_node
+            # for subgraphs with more than one node, start node != end_node
+            cur_start_node = cur_end_node
+
+            # Check for potential fusions. For now, we are greedy
+            # and always skip all non-base nodes of a fusion.  For example,
+            # if we match linear-relu backwards, we will always skip the
+            # relu node and attempt to match the linear node.  This can
+            # be made configurable later if needed.
+            for _reverse_fusion_ops in get_reversed_fusions():
```

Comment:
Do we always pick the largest fusion here?

Reply:
Ideally the ordering in the default would be followed, with the ability for user to customize.  It doesn't handle that yet.  We'd have to add that logic as soon as we have more than one fusion.